# on texgen 3.14 there is a explicit dependency on libraries saved in a dir -> removed that
diff --git a/Core/CMakeLists.txt b/Core/CMakeLists.txt
index 88c3094..2b962e6 100644
--- a/Core/CMakeLists.txt
+++ b/Core/CMakeLists.txt
@@ -48,13 +48,20 @@ ENDIF(WIN32)
 
 TARGET_INCLUDE_DIRECTORIES(TexGenCore PUBLIC ../OctreeRefinement/include)
 
-IF(UNIX)
-TARGET_LINK_LIBRARIES(TexGenCore triangle triangle-api Octree TinyXML CSparse tetgenlib
-${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libp4est-2.0.so
-${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libp4est.so
-${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libsc-2.0.so
-${CMAKE_CURRENT_SOURCE_DIR}/../OctreeRefinement/libsc.so)
-ENDIF(UNIX)
+if(UNIX)
+    find_package(SC REQUIRED)
+    find_package(P4EST REQUIRED)
+    target_link_libraries(TexGenCore
+        triangle
+        triangle-api
+        Octree
+        TinyXML
+        CSparse
+        tetgenlib
+        SC::SC
+        P4EST::P4EST
+    )
+endif()
 
 INCLUDE(../Python/FindPythonSitePackages.cmake)
 IF(WIN32)
